cli/internal/otel/receiver.go

```diff
@@ -242,19 +272,24 @@ func WithMetricsFilename(filename string) OtelReceiverOption {
 
 func StartOtelReceiver(ctx context.Context, opts ...OtelReceiverOption) (*OtelReceiver, error) {
```

Comment:
Maybe a cleaner API would be to pass an additional option to `StartOtelReceiver` where the option is a callback that gets invoked on each time we get a new duration metric?

Reply:
Good idea, done in https://github.com/cloudquery/cloudquery/pull/21335/commits/7fc42cebe2fec2f469acb45104b9dc857d7111ef